### ISSUE TITLE
[问题/Issue] 章节5.3.2：没有在结尾添加EOS Token / Chapter5.3.2: Missing the padding for EOS Token

### ISSUE THREAD

**Issue by xiaolin9595** [datawhalechina/happy-llm#33]
### 1. 遇到问题的章节 / Affected Chapter

Chapter5.3.2

### 2. 具体问题描述 / Problem Description

I noticed that in the __getitem__ method, the code only adds a BOS token to the beginning of the text, but does not add an EOS token at the end. For many language modeling tasks, adding an EOS token helps the model to learn the end of the sequence and improves generation quality.

Is there a specific reason for not adding the EOS token here? Would it be better to explicitly append the EOS token to the text before tokenization?

### 3. 问题重现材料 / Reproduction Materials

from torch.utils.data import Dataset

class PretrainDataset(Dataset):
    def __init__(self, data_path, tokenizer, max_length=512):
        super().__init__()
        self.data_path = data_path
        self.tokenizer = tokenizer
        self.max_length = max_length
        self.padding = 0
        with open(data_path, 'r', encoding='utf-8') as f:
            self.data = f.readlines()

    def __len__(self):
        return len(self.data)

    def __getitem__(self, index: int):
        sample = json.loads(self.data[index])
        text = f"{self.tokenizer.bos_token}{sample['text']}"
        input_id = self.tokenizer(text).data['input_ids'][:self.max_length]
        text_len = len(input_id)
        # 没满最大长度的剩余部分
        padding_len = self.max_length - text_len
        input_id = input_id + [self.padding] * padding_len
        # 0表示不计算损失
        loss_mask = [1] * text_len + [0] * padding_len

        input_id = np.array(input_id)
        X = np.array(input_id[:-1]).astype(np.int64)
        Y = np.array(input_id[1:]).astype(np.int64)
        loss_mask = np.array(loss_mask[1:]).astype(np.int64)
        return torch.from_numpy(X), torch.from_numpy(Y), torch.from_numpy(loss_mask)

### 确认事项 / Verification

- [x] 此问题未在过往Issue中被报告过 / This issue hasn't been reported before

**Comment by KMnO4-zx** [comment]
请看 issue18 问题2

https://github.com/datawhalechina/happy-llm/issues/18
